### DISCUSSION TITLE
Question about the RustDesk architecture

### DISCUSSION THREAD

**Discussion by foomoto** [rustdesk/rustdesk#7791] [General]
Amazing work everyone. I was looking to understand how rust desk works to take this understanding into a similar project I am starting. 

My understand is that RustDesk is installed on the both clients (the target device), and the remote agent. Is my understanding correct that when the remote agent controls the device, the mouse events are streamed over air and programmatically executed by the rust client on the target device?

If so, can you point me to the folder where all programmatic mouse movements and keyboard events are triggered

Thanks

**Comment by C-Duv** [comment]
Did you find out what you need?

I'm also looking for insights on  how it works on the computer: which part/process handles the heavy work (networking, registering with the server, etc.) and which one handles the GUI (settings window, connection manager, chat, etc.), in a multi-user OS.
